src/Common/src/TypeSystem/IL/ILImporter.cs

```diff
@@ -471,9 +474,16 @@ private void ImportBasicBlock(BasicBlock basicBlock)
                             ImportBranch(opCode,
                                 _basicBlocks[_currentOffset + delta], (opCode != ILOpcode.br) ? _basicBlocks[_currentOffset] : null);
                         }
+                        EndImportingInstruction();
                         return;
                     case ILOpcode.switch_:
                         {
+                            if (_currentOffset + 3 >= _ilBytes.Length)
```

Comment:
Should these cases rather be handled in `FindJumpTargets` ? We will hit some of them there at least.

Reply:
You are right, FindJumpTargets will be the first to detect them. We should handle them there, thanks!

I think it would make sense to abort the method verification, when such an error is reported. Otherwise this will cause either multiple reports or a crash in a later part (e.g. `ImportBasicBlock`, `InitialPass`, etc.).